modules/stb/0.0.0-20241109-5c20573/overlay/BUILD.bazel

```diff
@@ -0,0 +1,22 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "stb_image",
+    hdrs = ["stb_image.h"],
+    defines = ["STB_IMAGE_IMPLEMENTATION"],
```

Comment:
Based on https://github.com/nothings/stb?tab=readme-ov-file#how-do-i-use-these-libraries, wouldn't all targets need to look like this?

```starlark
cc_library(
    name = "stb_thing",
    hdrs = ["stb_thing.h"],
    # A generated file that includes the header after defining `STB_THING_IMPLEMENTATION`.
    srcs = ["stb_thing.c"],
    ...
)
```

This would allow users to use the headers without having to worry about the "include in exactly one source file" bit. You could probably get away with an empty source file if you use `local_defines` to set the macro and `-include` to force inclusion of the header.